elastica/boundary_conditions.py

```diff
@@ -401,35 +481,100 @@ def nb_constrain_translational_rates(velocity_collection, indices) -> None:
             2D (dim, blocksize) array containing data with `float` type.
         indices : numpy.ndarray
             1D array containing the index of constraining nodes
+        constraint_selector: numpy.ndarray
+            1D array of type int and size (3,) indicating which translational Degrees of Freedom (DoF) to constrain.
+            Entries are integers in {0, 1} (e.g. a binary values of either 0 or 1).
+            If entry is 1, the concerning DoF will be constrained, otherwise it will be free for translation.
+            Selector shall be specified in the inertial frame
         """
 
         block_size = indices.size
         for i in range(block_size):
             k = indices[i]
-            velocity_collection[0, k] = 0.0
-            velocity_collection[1, k] = 0.0
-            velocity_collection[2, k] = 0.0
+            velocity_collection[..., k] = (
+                1 - constraint_selector
+            ) * velocity_collection[..., k]
 
     @staticmethod
     @njit(cache=True)
-    def nb_constrain_rotational_rates(omega_collection, indices) -> None:
+    def nb_constrain_rotational_rates(
+        director_collection, omega_collection, indices, constraint_selector
+    ) -> None:
         """
         Compute constrain rates in numba njit decorator
 
         Parameters
         ----------
+        director_collection : numpy.ndarray
+            2D (dim, blocksize) array containing data with `float` type.
         omega_collection : numpy.ndarray
             2D (dim, blocksize) array containing data with `float` type.
         indices : numpy.ndarray
             1D array containing the index of constraining nodes
+        constraint_selector: numpy.ndarray
+            1D array of type int and size (3,) indicating which rotational Degrees of Freedom (DoF) to constrain.
+            Entries are integers in {0, 1} (e.g. a binary values of either 0 or 1).
+            If an entry is 1, the rotation around the respective axis will be constrained,
+            otherwise the system can freely rotate around the axis.
+            Selector shall be specified in the inertial frame
         """
+        directors = director_collection[..., indices]
 
-        block_size = indices.size
-        for i in range(block_size):
-            k = indices[i]
-            omega_collection[0, k] = 0.0
-            omega_collection[1, k] = 0.0
-            omega_collection[2, k] = 0.0
+        # rotate angular velocities to inertial frame
+        omegas_inertial = _batch_matvec(
+            directors.transpose(1, 0, 2), omega_collection[..., indices]
+        )
+
+        # apply constraint selector to angular velocities in inertial frame
+        omegas_allowed = (1 - np.expand_dims(constraint_selector, 1)) * omegas_inertial
```

Comment:
```suggestion
        omega_collection_not_constrained = (1 - np.expand_dims(constraint_selector, 1)) * omegas_inertial
```

Reply:
Fixed in https://github.com/GazzolaLab/PyElastica/pull/143/commits/2d135846f77908b927bbbb6bfe54daf760f7d9d7

Reply:
Changed in https://github.com/GazzolaLab/PyElastica/pull/143/commits/970cf0247d154e772bcd9dd861e1e7fc2071bcf8